src/Expander/Task.hs

```diff
@@ -23,7 +23,8 @@ import Value
 data MacroDest
   = ExprDest Ty SplitCorePtr
   | TypeDest SplitTypePtr
-  | DeclDest DeclPtr Scope DeclValidityPtr
+  | DeclTreeDest DeclTreePtr ScopeSet DeclValidityPtr
+    -- ^ tree node to fill, input environment, output environment
```

Comment:
Added a comment to further drive home that the `ScopeSet` / first `DeclValidityPtr` is always the input environment while the second `DeclValidityPtr` is the output environment.

Reply:
Good! By "environment" you mean "scopes to add to these decls" vs "scopes to add to subsequent decls", right?

Reply:
To subsequent decls. The input `ScopeSet` is the set of `Scopes` which the previous decls have created (they have already been added to the `Syntax` in `forkExpandDecls`), and the output `DeclValidityPtr` must be filled with a `ScopeSet` which includes both the input `ScopeSet` and the `Scope`s created by the code which fills this `DeclTreePtr`, so that subsequent decls can see the identifiers bound by that code.

Reply:
I have changed the comment to avoid misusing the word "environment". The comment is much longer now; maybe too long?